cbor.js

```diff
@@ -50,12 +141,53 @@ function encode(value) {
     lastLength = length;
     return dataView;
   }
+
+  function maybeIntKey(key) {
+    var charCode = key.charCodeAt(0);
+    var isNumKey = charCode >= 48 && charCode <= 57;
+    for (var ki = 1; isNumKey && ki < key.length; ++ki) {
+      charCode = key.charCodeAt(ki);
+      isNumKey = charCode >= 48 && charCode <= 57;
+    }
+    if (isNumKey)
+      key = parseInt(key, 10);
+    return key;
+  }
+
   function commitWrite() {
     offset += lastLength;
   }
-  function writeFloat64(value) {
-    commitWrite(prepareWrite(8).setFloat64(offset, value));
+
+  function writeFloat16(value) {
+    dataView.setUint8(offset, 0xf9);
+    dataView.setUint16(offset+1, getFloat16(value));
+    offset += 3;
+  }
+
+  function writeFloat(value) {
```

Comment:
suffers from some artifacts of prepareWrite, commitWrite, writeFloat16, writeFloat32, and writeFloat64